code/core/src/manager-api/lib/open-in-editor.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  OPEN_IN_EDITOR_REQUEST,
+  OPEN_IN_EDITOR_RESPONSE,
+  type OpenInEditorResponsePayload,
+} from 'storybook/internal/core-events';
+
+import { addons } from './addons';
+
+export async function openInEditor(
+  file: string,
+  line?: number,
+  column?: number
+): Promise<OpenInEditorResponsePayload> {
+  return new Promise((resolve) => {
+    const channel = addons.getChannel();
+    const payload = { file, line, column };
```

Comment:
Why don't you just make `payload` the sole argument to the function? I think that's generally cleaner than 3 primitive arguments.